iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs

```diff
@@ -69,7 +69,7 @@ private void OnConnectionClosed(object o, EventArgs args)
 
             if (_amqpIotConnection != null && ReferenceEquals(_amqpIotConnection, o))
             {
-                _amqpAuthenticationRefresher?.StopLoop();
+                _ = _amqpAuthenticationRefresher?.StopLoopAsync().ConfigureAwait(false);
```

Comment:
This event handler cannot await on async code. This handler is generally invoked as a result of the "Close" call. Since the close call with stop and await the token refresher cancellation first, this is merely added as an extra level of check.